docs/docs/debugging-gatsby.md

```diff
@@ -0,0 +1,132 @@
+
+![Files added to Sources tab](./images/chrome-devtools-files.png)
+
+### Using DevTools
+
+Let's go ahead and add a breakpoint just before the place that the error is thrown. To add a breakpoint navigate to `gatsby-node.js` and left click on a line number:
+
+![Added breakpoint](./images/chrome-devtools-new-breakpoint.png)
+
+Now you can resume code execution by clicking the "resume" icon in the DevTools debug toolbar (or press F8 on your keyboard). Gatsby will start running and pause once it reaches breakpoint allowing you to inspect variables:
+
+![Breakpoint hit](./images/chrome-devtools-breakpoint-hit.png)
+
+To inspect variables you can hover your mouse over them or go to the `Scope` section in the right-hand pane (either collapse the "Call Stack" section or scroll through it to the bottom).
+
+In our example `Node` is `undefined` and to figure out why, let's go backwards. `Node` is extracted from `args` so let's examine that by hovering `args`:
+
+![Examine variable](./images/chrome-devtools-examine-var.png)
+
+We can now see the problem - `args` doesn't contain `Node` - it contains `node`. So this small typographic mistake was causing our code to fail. Adjusting our code to use a lowercase `node` fixes the problem and we did that without adding tons of `console.log` output!
+
+### Finishing thoughts
+
+We can succussfully debug our code using Chrome DevTools but using it isn't really that convenient. There are a lot of steps we need to do manually everytime we want to use debugger, so in the next section you'll learn how to use the built-in debugging capabilities of VS Code.
+
+## VS Code debugger
+
+Using built in debuggers in code editors is very convenient. You will be able to skip a lot of setup needed to use Chrome DevTools. You also will be able to put breakpoints in same view you write your code.
+
+We won't go in depth here about how to debug in VS Code - for that you can check the [excellent VS Code documentation](https://code.visualstudio.com/docs/editor/debugging). We will however share a launch configuration needed to run and debug Gatsby:
+
+`launch.json`:
+```json
+{
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "name": "Gatsby develop",
+      "type": "node",
+      "request": "launch",
+      "protocol": "inspector",
+      "program": "${workspaceRoot}/node_modules/gatsby/dist/bin/gatsby",
+      "args": ["develop"],
+      "stopOnEntry": false,
+      "runtimeArgs": ["--nolazy"]
+    },
+    {
+      "name": "Gatsby build",
+      "type": "node",
+      "request": "launch",
+      "protocol": "inspector",
+      "program": "${workspaceRoot}/node_modules/gatsby/dist/bin/gatsby",
+      "args": ["build"],
+      "stopOnEntry": false,
+      "runtimeArgs": ["--nolazy"]
+    }
+  ]
+}
+```
+
+After putting a breakpoint in `gatsby-node.js` and using the `Start debugging` command from VS Code you can see the final result:
+
+![VSCode breakpoint hit](./images/vscode-debug.png)
```

Comment:
Maybe there could be a summary at the end that links to the "how to contribute" docs, and maybe some other good resources about debugging?